### PR TITLE
Fix Clustername Modification

### PR DESCRIPTION
**Pull request by nemonster**
* Fixed issue with cluster name not being updated in imported indices
* Fixed validation issue with input clustername
* Modified Service Objects in export/import to be threadsafe.

### CODE REVIEW

File: src/main/java/com/elastic/support/monitoring/MonitoringExportApp.java [hunk 1]
```diff
@@ -17,7 +17,6 @@ public static void main(String[] args) {
             System.exit(0);
         }
 
-        MonitoringExportService service = new MonitoringExportService(monitoringExportInputs);
-        service.execExtract();
+        new MonitoringExportService().execExtract(monitoringExportInputs);
```
Comment (pickypg) [inline src/main/java/com/elastic/support/monitoring/MonitoringExportApp.java]: Seems like this method makes more sense as a `static` one now that its only attribute is the logger?

File: src/main/java/com/elastic/support/monitoring/MonitoringExportService.java [hunk 2]
```diff
@@ -28,24 +28,24 @@
...
 
-    public void execExtract() {
+        RestClient client = null;
+        Map configMap = JsonYamlUtils.readYamlFromClasspath(Constants.DIAG_CONFIG, true);
+        MonitoringExportConfig config = new MonitoringExportConfig(configMap);
 
         try {
+            client = createEsRestClient(config, inputs);
+
+            String tempDir = "";
```
Comment (pickypg) [inline src/main/java/com/elastic/support/monitoring/MonitoringExportService.java]:
```suggestion
            String tempDir = SystemProperties.fileSeparator + Constants.MONITORING_DIR;
```

File: src/main/java/com/elastic/support/monitoring/MonitoringExportService.java [hunk 3]
```diff
@@ -28,24 +28,24 @@
...
+            if(StringUtils.isNotEmpty(inputs.getOutputDir())){
+                tempDir = inputs.getOutputDir() + SystemProperties.fileSeparator + Constants.MONITORING_DIR;;
```
Comment (pickypg) [inline src/main/java/com/elastic/support/monitoring/MonitoringExportService.java]:
```suggestion
                tempDir = inputs.getOutputDir() + tempDir;
```

File: src/main/java/com/elastic/support/monitoring/MonitoringExportService.java [hunk 4]
```diff
@@ -28,24 +28,24 @@
...
+            }
+            else{
+                tempDir = SystemProperties.userDir + SystemProperties.fileSeparator + Constants.MONITORING_DIR;
```
Comment (pickypg) [inline src/main/java/com/elastic/support/monitoring/MonitoringExportService.java]:
```suggestion
                tempDir = SystemProperties.userDir + tempDir;
```

File: src/main/java/com/elastic/support/monitoring/MonitoringExportApp.java (hunk 1 above)
Comment (nemonster) [reply src/main/java/com/elastic/support/monitoring/MonitoringExportApp.java]: You could actually say that about a number of the current classes. But I don't want to do that yet - I still may change my mind.

**Pull request merged**